### PR TITLE
Adds support for bitbucket cloud

### PR DESCRIPTION
**Pull request by WolfgangD**
- Duplicates bitbucket_server RequestSource and Plugin and adapts them for bitbucket cloud REST API.
- Dangerfile support is the same as for bitbucket_server but uses bitbucket_cloud name.
- Basic Auth via username/password is used and needs to be set via DANGER_BITBUCKETCLOUD_USERNAME and DANGER_BITBUCKETCLOUD_PASSWORD environment vars

### CODE REVIEW

File: lib/danger/request_source/bitbucket_cloud.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+        comment = generate_description(warnings: warnings, errors: errors)
+        comment += "\n\n"
+        comment += generate_comment(warnings: warnings,
+                                     errors: errors,
+                                   messages: messages,
+                                  markdowns: markdowns,
+                        previous_violations: {},
+                                  danger_id: danger_id,
+                                   template: "bitbucket_server")
```
Comment (JuanitoFatas) [inline lib/danger/request_source/bitbucket_cloud.rb]: `bitbucket_cloud` 😊 ?

Comment (WolfgangD) [inline lib/danger/request_source/bitbucket_cloud.rb]: if I understood this correctly this is the template for the comment that get's posted to bitbucket. As the systems should be similar and the comment looks okay; I decided to stay with the same template.

Comment (orta) [inline lib/danger/request_source/bitbucket_cloud.rb]: This makes sense

File: lib/danger/request_source/bitbucket_cloud.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      def initialize(ci_source, environment)
+        self.ci_source = ci_source
+        self.environment = environment
+
+        project, slug = ci_source.repo_slug.split("/")
+        @api = BitbucketCloudAPI.new(project, slug, ci_source.pull_request_id, environment)
+      end
+
+      def validates_as_ci?
+        # TODO: ???
```
Comment (orta) [inline lib/danger/request_source/bitbucket_cloud.rb]: I love how this comment of mine is propagating itself around the system 👯

**Comment by JuanitoFatas** [comment]
LGTM, this is so cool 😎

**Comment by orta** [comment]
Yep, looks all 👍

**Comment by orta** [comment]
@WolfgangD would you like to join the org?

**Pull request merged**